test/unit/foreman/renderer_test.rb

```diff
@@ -44,18 +44,15 @@ class RendererTest < ActiveSupport::TestCase
   private
 
   def assert_template(template)
-    rendered = Foreman::TemplateSnapshotService.render_template(template)
-    variants = Foreman::Renderer::Source::Snapshot.snapshot_variants(template)
-    match = variants.any? { |variant| rendered == File.read(variant) }
-
-    # print diff against all compared files
-    unless match
-      variants.each do |variant|
-        puts "Diff for #{variant}:"
-        puts diff(File.read(variant), rendered)
+    Foreman::Renderer::Source::Snapshot.hosts(template).each do |host|
+      snapshot_path = Foreman::Renderer::Source::Snapshot.snapshot_path(template, host)
+      rendered = Foreman::TemplateSnapshotService.render_template(template, host)
+      unless rendered == File.read(snapshot_path)
+        puts "Diff for #{snapshot_path}:"
+        puts diff(File.read(snapshot_path), rendered)
+
+        assert false, "Rendered template #{template.name} did not match the snapshot."
       end
-
-      assert match, "Rendered template #{template.name} did not match any snapshot. Tried against #{variants.join(', ')}"
     end
   end
 end
```

Comment:
Here, @timogoebel I do not understand why rendered template payload is being matched against various "variants", so I am doing the comparison the same way we generate them in `snapshots.rake`. But please confirm, I have a vague memory that this was because a plugin?

Reply:
@kamils-iRonin: Do you mind chiming in here? Thanks.

Reply:
it looks that variants are a way to override snapshots in the plugin tests (see https://github.com/theforeman/foreman/commit/8413b711395388bbf9826606ae0ae58f1f2ec64e)